Libs/Gst/Structs/Message.cs

```diff
@@ -0,0 +1,141 @@
+            oldState = (State)oldStatePtr;// != IntPtr.Zero ? Marshal.PtrToStructure<State>(oldStatePtr) : null;
+            newState = (State)newStatePtr;// != IntPtr.Zero ? Marshal.PtrToStructure<State>(newStatePtr) : null;
+            pendingState = (State)pendingStatePtr;// != IntPtr.Zero ? Marshal.PtrToStructure<State>(pendingStatePtr) : null;
+        }
+
+        public void ParseTag(out TagList? tagList)
+        {
+            if (type != MessageType.Tag)
+                throw new MessageTypeMismatchException(type, MessageType.Tag);
+            
+            // Empty pointers
+            IntPtr tagListPtr = default;
+
+            // Marshal this structure
+            IntPtr thisPtr = Marshal.AllocHGlobal(Marshal.SizeOf(this));
+            Marshal.StructureToPtr(this, thisPtr, false);
+            
+            // Native Call
+            Native.parse_tag(thisPtr, out tagListPtr);
+
+            // Update this structure (TODO: Check for NULL)
+            this = (Message)Marshal.PtrToStructure(thisPtr, GetType())!;
+            
+            // Free Memory
+            Marshal.FreeHGlobal(thisPtr);
+
+            // Assign out variables
+            tagList = tagListPtr != IntPtr.Zero ? Marshal.PtrToStructure<TagList>(tagListPtr) : null;
+        }
+        
+        public void ParseBuffering(out int percent)
+        {
+            if (type != MessageType.Buffering)
+                throw new MessageTypeMismatchException(type, MessageType.Buffering);
+            
+            // Empty pointers
+            percent = 0;
+
+            // Marshal this structure
+            IntPtr thisPtr = Marshal.AllocHGlobal(Marshal.SizeOf(this));
+            Marshal.StructureToPtr(this, thisPtr, false);
+            
+            // Native Call
+            Native.parse_buffering(thisPtr, out percent);
+
+            // Update this structure (TODO: Check for NULL)
+            this = (Message)Marshal.PtrToStructure(thisPtr, GetType())!;
+            
+            // Free Memory
+            Marshal.FreeHGlobal(thisPtr);
+        }
+
+        public void ParseError(out GLib.Error? error, out string? debug)
+        {
+            if (type != MessageType.Error)
+                throw new MessageTypeMismatchException(type, MessageType.Error);
+            
+            // Empty pointers
+
+            // Marshal this structure
+            IntPtr thisPtr = Marshal.AllocHGlobal(Marshal.SizeOf(this));
+            Marshal.StructureToPtr(this, thisPtr, false);
+
+            // Native Call
+            Native.parse_error(thisPtr, out IntPtr errPtr, out IntPtr strPtr);
+
+            // Update structures (TODO: Check for NULL)
+            this = (Message) Marshal.PtrToStructure(thisPtr, GetType())!;
+            debug = Marshal.PtrToStringAnsi(strPtr);
+            error = Marshal.PtrToStructure<GLib.Error>(errPtr);
+
+            // Free Memory
+            Marshal.FreeHGlobal(thisPtr);
+        }
+
+        public Gst.Object Src
```

Comment:
Properties should be at the top of the file.

Reply:
Fixed